### PR TITLE
feat: read commit message from env var

### PR DESCRIPTION
**Pull request by dschmidt**
The `message` setting accepts a static string or a Woodpecker-substituted `${CI_COMMIT_MESSAGE}`. Substitution happens textually before YAML parsing, so messages containing characters that are meaningful to the YAML parser break the pipeline:

- A literal `:` (every conventional commit, e.g. `fix(api): …`) triggers a `mapping values are not allowed in this context` parse error.
- Quoting around the substitution (`message: "${CI_COMMIT_MESSAGE}"`) handles `:` but breaks on a literal `"` in the message. Single-quoting has the same flaw for `'`.

This PR adds an opt-in `commit-message-from` setting that names an environment variable from which the plugin reads the commit message at runtime, completely bypassing the YAML substitution path:

```yaml
settings:
  action: [commit, push]
  commit-message-from: CI_COMMIT_MESSAGE
```

Semantics:

- If `commit-message-from` is set and the named env var resolves to a non-empty value, that value is used. It overrides both an explicit `message` setting and the static default.
- Otherwise (`commit-message-from` unset, or env var unset/empty), `message` is used as today — either its explicit value or the static default `"[skip ci] commit dirty state"`.

The existing `Sources: cli.EnvVars("PLUGIN_AUTHOR_NAME", "CI_COMMIT_AUTHOR")` fallback pattern for `author_name`/`author_email` was deliberately not used here for `message`: doing so would silently shadow the static `"[skip ci] commit dirty state"` default in any Woodpecker context (where `CI_COMMIT_MESSAGE` is always set) and would be a breaking change to the plugin's defaults.

### Motivation

In [opencloud-eu/libre-graph-api#48](https://github.com/opencloud-eu/libre-graph-api/pull/48) we hit the YAML substitution problem on multiple pipelines and currently work around it by quoting `${CI_COMMIT_MESSAGE}` in YAML. That fix only covers `:`, not `"`; we kept it as the lesser evil but would much rather drop the workaround entirely with a clean plugin-level solution. With this PR, the consumer pipeline becomes simply `commit-message-from: CI_COMMIT_MESSAGE`.

### Changes

- New `commit-message-from` string flag (`PLUGIN_COMMIT_MESSAGE_FROM`).
- `Validate()` overrides `Repo.CommitMsg` from the named env var when it resolves to a non-empty value.
- Unit tests covering all four branches (flag unset / env empty falls back to message / env populated overrides default / env populated overrides explicit message).
- Docs entry in `docs/data/data.yaml`.

### CODE REVIEW

**Comment by xoxys** [comment]
🍻

File: plugin/impl.go [hunk 1]
```diff
@@ -50,6 +50,16 @@ func (p *Plugin) Validate() error {
 	p.Settings.Repo.Autocorrect = "never"
 	p.Settings.Repo.RemoteName = "origin"
 
+	// If commit-message-from names an environment variable that resolves to a
+	// non-empty value, adopt that value as the commit message. The 'message'
+	// setting (explicit or default) acts as a fallback when the env var is
+	// unset or empty.
+	if p.Settings.CommitMessageFrom != "" {
+		if v := os.Getenv(p.Settings.CommitMessageFrom); v != "" {
```
Comment (xoxys) [inline plugin/impl.go]: Use `strings.TrimSpace` here before the comparison.

File: plugin/impl_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	tests := []struct {
+		name              string
+		commitMessageFrom string
+		envValue          string
+		initialMsg        string
+		wantMsg           string
+	}{
+		{
+			name:              "from-flag unset leaves default",
```
Comment (xoxys) [inline plugin/impl_test.go]:
```suggestion
			name:              "commit-message-from unset leaves default",
```

same for all other test cases

File: plugin/impl_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			initialMsg:        staticDefault,
+			wantMsg:           "feat(api): add motion photo facet",
+		},
+		{
+			name:              "from-flag set and env populated overrides explicit message",
+			commitMessageFrom: ciVar,
+			envValue:          "from CI",
+			initialMsg:        "explicit message",
+			wantMsg:           "from CI",
+		},
```
Comment (xoxys) [inline plugin/impl_test.go:48]: Nit: Lets add a small test that we ignore `envValue` which contains only whitespace chars.

**Review by xoxys — COMMENTED** [review]
Thanks! Only a few nits.

File: plugin/impl.go (hunk 1 above)
Comment (dschmidt) [reply plugin/impl.go]: Done in a01cf25.

File: plugin/impl_test.go (hunk 2 above)
Comment (dschmidt) [reply plugin/impl_test.go]: Done in a01cf25.

File: plugin/impl_test.go (hunk 3 above)
Comment (dschmidt) [reply plugin/impl_test.go:48]: Done in a01cf25.

**Comment by xoxys** [comment]
Can you check the linting error from last CI run?

**Comment by dschmidt** [comment]
done

**Comment by dschmidt** [comment]
🍺 🍺, 🍺 🍺 , 🧔‍♂️ 🥳 - 🧔‍♂️ 🍺 🤖
